### PR TITLE
[script.module.future] 0.17.1

### PR DESCRIPTION
**Pull request by Freso**
### Description

* Update [future](https://python-future.org/) Python module to 0.17.1. See [upstream `whatsnew.rst`](https://github.com/PythonCharmers/python-future/blob/master/docs/whatsnew.rst) for upstream changes.

The [0.16.0.4 PR](https://github.com/xbmc/repo-scripts/pull/761) changed the upstream package somewhat for case-sensitive platforms. AFAICT those same issues do not affect 0.17.x, so I haven’t attempted at porting any of those alterations over—but I also can’t test on non‐Linux platforms (incl. Android) to verify, so I’d especially appreciate a review from @romanvm. :)

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Freso** [comment]
I feel like the given error might be "to be expected"? I don’t know if it actually runs the code or just checks to see if it’s valid Python (2 &) 3 code.

**Comment by romanvm** [comment]
Do I understand correctly that this just a port of the upstream `future` package with no changes?

**Comment by Freso** [comment]
Yep. Straight `tar -xzvf future-0.17.1.tar.gz`’ing (and moving some directories out of the way). No changes to upstream code.

File: script.module.future/libs/future/standard_library/__init__.py [hunk 1]
```diff
@@ -79,17 +79,6 @@
 
 from future.utils import PY2, PY3
 
-# Added by Roman V.M. for using in Kodi
```
[the remark this answers is not in the document]
Comment (Freso) [reply script.module.future/libs/future/standard_library/__init__.py]: It turns out my local file tree had gotten messed up somehow, making me miss a bunch of files, which led me to believe this was no longer needed. After resetting and going through the motions again, I realised my assumption was mistaken. I’ve pushed a new commit that includes this blurb again.

**Comment by Freso** [comment]
Seems the latest Travis run failed because of a connection error, and I don’t have permissions here to make it re‐run. (Also, is there a way to stop @TravisBuddy from leaving dozens of comments?)

**Comment by Freso** [comment]
Looks like there’s an error in `kodi_addon_checker/check_py3_compatibility.py` – or at least there’s something that triggers an exception that makes it not report useful output for actually fixing if there’s anything wrong in the PR code.

**Comment by romanvm** [comment]
@Freso Have you actually tested this in Kodi - both py2 and py3 builds? If yes then it should be merged. Unfortunately, I can't tests it myself at the moment.

**Comment by Freso** [comment]
I can’t test on py3 builds, but I’ve been using this in my Kodi 18.1 install without any issues. There’s also nothing in the upstream changelog that suggests any backwards incompatible changes. Is there any specific testing you have in mind?

**Comment by enen92** [comment]
good to go?

**Pull request merged**